engine/engine.go

```diff
@@ -176,7 +176,14 @@ func (e *engine) loadDB() (err error) {
 
 	m, err := migrate.NewWithDatabaseInstance("file://migration/", "sqlite3", driver)
 	if err != nil {
-		return fmt.Errorf("initializing db migration failed %s", err)
+		// When running tests, we might be inside the engine directory.
```

Comment:
I am not sure if we should ever modify the actual code to accommodate tests in this manner, as a guiding principal. I haven't seen a case where we must do it. So far, I have been able to completely avoid it.